### PR TITLE
feat: markasread in dropdown

### LINKED ISSUE #3313: feat: Include mark as read button in the drop down menu of notifications
**Is your feature request related to a problem? Please describe.**
I think it will be much more user friendly if users had an option to mark all notifications as read from the drop down itself ... rather than going to the notification page every time. 

**Describe the solution you'd like**
To include 
- mark as read button in the drop down menu of notifications

I propose a solution like the one mentioned below ...
**Describe alternatives you've considered**

**Additional context**

BEFORE 
before read

AFTER (no more notifications)
after read

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by VaibhavUpreti**
Fixes #3313 

#### Describe the changes you have made in this PR -
1. Added mark as read link in notification drop down 

### Screenshots of the changes (If any) -
BEFORE FIX
beforefix

AFTER FIX
afterfix

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -11,7 +11,6 @@
 # It's strongly recommended that you check this file into your version control system.
 
 ActiveRecord::Schema[7.0].define(version: 2022_08_13_051332) do
```
Comment (tachyons) [inline db/schema.rb:13]: This change is unrelated, please remove from PR

**Comment by VaibhavUpreti** [comment]
@tachyons Removed the unwanted commits , please have a look. Cheers 👍

**Comment by VaibhavUpreti** [comment]
Unread notifications-
Screenshot 2022-11-01 at 11 13 24 PM
When no unread notifications-

Screenshot 2022-11-01 at 11 14 01 PM

**Comment by VaibhavUpreti** [comment]
Fixes issue #3313 .. @tachyons @vedant-jain03 kindly review this.

File: app/views/layouts/_notifications.html.erb [hunk 2]
```diff
@@ -38,7 +38,12 @@
         </div>
       <% end %>
       <div class="d-flex align-items-center justify-content-center mb-2 mt-3">
+        <% if @unread.count.positive? %>
+          <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text-left" %>
```
Comment (vedant-jain03) [inline app/views/layouts/_notifications.html.erb:42]: This will redirect the user to the `notification` page, but if the user clicks on `mark all as read` from a different page it should stick to that page only!

You need to write another method, route for this one @VaibhavUpreti

Comment (VaibhavUpreti) [reply app/views/layouts/_notifications.html.erb]: Added another route and defined another method ... now after reading notifications from the modal user sticks to the same page.

**Comment by VaibhavUpreti** [comment]
@tachyons @vedant-jain03 is this good for mergre?

**Review by vedant-jain03 — APPROVED** [review]

File: config/routes.rb [hunk 3]
```diff
@@ -80,6 +80,7 @@
     get "/educational_institute/typeahead/:query" => "users/circuitverse#typeahead_educational_institute"
     get "/:id/notifications", to: "users/noticed_notifications#index", as: "notifications"
     get "/:id/notifications/mark_all_as_read", to: "users/noticed_notifications#mark_all_as_read", as: "mark_all_as_read"
+    get "/:id/notifications/read_all_notifications", to: "users/noticed_notifications#read_all_notifications", as: "read_all_notifications"
```
Comment (tachyons) [inline config/routes.rb:83]: both of these should be `PATCH` not `GET`

**Comment by VaibhavUpreti** [comment]
@vedant-jain03 @tachyons updated "get" to "patch" ... also had to change one spec 👍  ... all tests passing
